core/Updater.php

```diff
@@ -83,14 +83,52 @@ public function addUpdateObserver(UpdateObserver $listener)
      *
      * @param string $name The component name. Eg, a plugin name, `'core'` or dimension column name.
      * @param string $version The component version (should use semantic versioning).
+     * @param bool   $isNew indicates if the component is a new one (for plugins)
      */
-    public function markComponentSuccessfullyUpdated($name, $version)
+    public function markComponentSuccessfullyUpdated($name, $version, $isNew = false)
     {
         try {
             Option::set(self::getNameInOptionTable($name), $version, $autoLoad = 1);
         } catch (\Exception $e) {
             // case when the option table is not yet created (before 0.2.10)
         }
+
+        if ($isNew) {
+
+            /**
+             * Event triggered after a new component has been installed.
+             *
+             * @param string $name The component that has been installed.
+             */
+            Piwik::postEvent('Updater.componentInstalled', array($name));
+
+            return;
+        }
+
+        /**
+         * Event triggered after a component has been updated.
+         *
+         * Can be used to handle stuff that should be done after a component was updated
+         *
+         * **Example**
+         *
+         *     Piwik::addAction('Updater.componentUpdated', function ($componentName, $updatedVersion) {
+         *          $mail = new Mail();
+         *          $mail->setDefaultFromPiwik();
+         *          $mail->addTo('test@example.org');
+         *          $mail->setSubject('Component was updated);
+         *          $message = sprintf(
+         *              'Component %1$s has been updated to version %2$s',
+         *              $componentName, $updatedVersion
+         *          );
+         *          $mail->setBodyText($message);
+         *          $mail->send();
+         *     });
+         *
+         * @param string $componentName 'core', or plugin name
```

Comment:
I think it could be also the name of a dimension etc. as they are basically components as well